services-js/registry-certs/client/birth/questions/BornInBoston.tsx

```diff
@@ -1,34 +1,107 @@
 import React from 'react';
 
+import QuestionComponent from './QuestionComponent';
 import FieldsetComponent from './FieldsetComponent';
 import YesNoUnsureComponent from './YesNoUnsureComponent';
 
 import { YesNoUnknownAnswer } from '../QuestionsFlow';
 
 interface Props {
   forSelf: boolean | null;
-  currentValue: YesNoUnknownAnswer;
+  bornInBoston?: YesNoUnknownAnswer;
+  parentsLivedInBoston?: YesNoUnknownAnswer;
 
-  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleProceed: (answers: Partial<State>) => void;
   handleStepBack: () => void;
+  handleUserReset: () => void;
 }
 
-export default function BornInBoston(props: Props): JSX.Element {
-  return (
-    <section>
-      <FieldsetComponent handleStepBack={props.handleStepBack}>
-        <legend>
-          <h2>
-            Were {props.forSelf ? 'you' : 'they'} born in the City of Boston?
-          </h2>
-        </legend>
-
-        <YesNoUnsureComponent
-          questionName="bornInBoston"
-          questionValue={props.currentValue}
-          handleChange={props.handleChange}
-        />
-      </FieldsetComponent>
-    </section>
-  );
+interface State {
+  bornInBoston: YesNoUnknownAnswer;
+  parentsLivedInBoston: YesNoUnknownAnswer;
+}
+
+export default class BornInBoston extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      bornInBoston: props.bornInBoston || '',
+      parentsLivedInBoston: props.parentsLivedInBoston || '',
+    };
+  }
+
+  private handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    } as any);
+  };
+
+  private allowProceed(): boolean {
+    if (this.state.bornInBoston === 'yes') {
+      return true;
+    } else if (
+      this.state.parentsLivedInBoston === 'yes' ||
+      this.state.parentsLivedInBoston === 'unknown'
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private startOver(): boolean {
```

Comment:
The name of this method threw me because it sounds like it's an action ("start the flow over again"), rather than a "should they start over?"